src/lib/alloc.c

```diff
@@ -555,36 +555,65 @@ static void *rmalloc_runtime(int zone, uint32_t caps, size_t bytes)
 	return get_ptr_from_heap(heap, zone, caps, bytes);
 }
 
+static void *rballoc(int zone, uint32_t caps, size_t bytes)
+{
+	void *ptr;
+
+	ptr = _balloc(zone, caps, bytes);
+
+#if CONFIG_DEBUG_HEAP
+	if (!ptr) {
+		trace_error(TRACE_CLASS_MEM,
+			    "failed to alloc 0x%x bytes caps 0x%x",
+			    bytes, caps);
+
+		alloc_trace_buffer_heap(zone, caps, bytes);
+	}
+#endif
+
+	return ptr;
+}
+
 /* allocates memory - not for direct use, clients use rmalloc() */
 void *_malloc(int zone, uint32_t caps, size_t bytes)
 {
 	uint32_t flags;
 	void *ptr = NULL;
 
+	if (bytes >= HEAP_BUFFER_BLOCK_SIZE)
```

Comment:
Looks like `RZONE_BUFFER` isn't used anymore.
Shouldn't it be used to take priority in allocation?
Some buffers explicitly want to be allocated directly on buffer heap and not on other heaps and some want to be allocated on system heaps. I don't see current code working with other cores running, where every core has its own system and system runtime heaps.
Also I think the current code will panic if you try to allocate buffer with `bytes < HEAP_BUFFER_BLOCK_SIZE` and zone `RZONE_BUFFER`.